arches/app/models/models.py

```diff
@@ -2152,3 +2163,47 @@ def __str__(self):
     class Meta:
         managed = True
         db_table = "spatial_views"
+
+    def clean(self):
+        """
+        Validate the spatial view before saving it to the database as the database triggers have proved hard to test.
+        """
+        graph = self.geometrynode.graph
+        for node in self.attributenodes:
+            if not Node.objects.filter(pk=node["nodeid"], graph=graph).exists():
+                raise ValidationError(
+                    f"Attribute nodes must belong to the same graph as the geometry node (error nodeid:{str(node['nodeid'])})"
+                )
+
+        # language must be be a valid language code belonging to the current publication
+        published_graphs = graph.publication.publishedgraph_set.all()
+        if self.language_id not in [
+            published_graph.language_id for published_graph in published_graphs
+        ]:
+            raise ValidationError(
+                "Language must belong to a published graph for the graph of the geometry node"
+            )
+
+        # validate the schema is a valid schema in the database
+        with connection.cursor() as cursor:
+            cursor.execute(
+                f"SELECT schema_name FROM information_schema.schemata WHERE schema_name = '{self.schema}'"
```

Comment:
I think this creates an SQL injection vector. You could use raw() to get a [nice interface](https://docs.djangoproject.com/en/4.2/topics/db/sql/#passing-parameters-into-raw) for paramterizing a raw sql query.

Reply:
Another good catch - now using params. I'm not querying a model so won't look at raw.